src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.ts

```diff
@@ -138,10 +138,17 @@ export default class AmazonPayPaymentStrategy implements PaymentStrategy {
         return amazon ? amazon.referenceId : undefined;
     }
 
+    private _getOrderReferenceIdFromOrder(): string | undefined {
+        const state = this._store.getState();
+        const order = state.order.getOrder();
+
+        return order ? order.providerTransactionId : undefined;
+    }
+
     private _createWallet(options: AmazonPayPaymentInitializeOptions): Promise<AmazonPayWallet> {
         return new Promise((resolve, reject) => {
             const { container, onError = noop, onPaymentSelect = noop, onReady = noop } = options;
-            const referenceId = this._getOrderReferenceId();
+            const referenceId = this._getOrderReferenceId() || this._getOrderReferenceIdFromOrder();
```

Comment:
What happens if both ar `undefined`?

Reply:
if `this._getOrderReferenceId()` comes undefined another `orderReferenceId` will be generated

Reply:
Okay, sound good.